mars/services/web/ui/src/task_visualization/TaskTileableChart.js

```diff
@@ -0,0 +1,147 @@
+                var value = { tileable };
+
+                let nameEndIndex = tileable.tileable_name.indexOf('key') - 1;
+                value.label = tileable.tileable_name.substring(0, nameEndIndex);
+                value.rx = value.ry = 5;
+                this.g.setNode(tileable.tileable_id, value);
+
+                // In future fill color based on progress
+                this.g.node(tileable.tileable_id).style = 'fill: #9fb4c2; cursor: pointer;';
+            });
+
+            this.state.dependencies.forEach((dependency) => {
+                // In future label may be named based on linkType?
+                this.g.setEdge(dependency.from_tileable_id, dependency.to_tileable_id, { label: '' });
+            });
+
+            let gInstance = this.g;
+            // Round the corners of the nodes
+            gInstance.nodes().forEach(function (v) {
+                var node = gInstance.node(v);
+                node.rx = node.ry = 5;
+            });
+
+            //makes the lines smooth
+            gInstance.edges().forEach(function (e) {
+                var edge = gInstance.edge(e.v, e.w);
+                edge.style = 'stroke: #333; fill: none';
+            });
+
+            // Create the renderer
+            var render = new DagRender();
+
+            // Set up an SVG group so that we can translate the final graph.
+            var svg = d3Select('#svg-canvas'),
+                inner = svg.append('g');
+
+            // Set up zoom support
+            var zoom = d3Zoom().on('zoom', function (e) {
+                inner.attr('transform', e.transform);
+            });
+            svg.call(zoom);
+
+            if (this.state.tileables.length !== 0) {
+                // Run the renderer. This is what draws the final graph.
+                render(inner, this.g);
+            }
+
+            const handleClick = (e, node) => {
+                const selectedTileable = this.state.tileables.filter((tileable) => tileable.tileable_id == node)[0];
+                this.setState({ selectedTileable: selectedTileable});
+            };
+
+            inner.selectAll('g.node').on('click', handleClick);
+
+            // Center the graph
+            var initialScale = 0.75;
+            svg.call(
+                zoom.transform,
+                d3ZoomIdentity.scale(initialScale)
+            );
+
+            if (this.g.graph() !== null || this.g.graph() !== undefined) {
+                svg.attr('height', this.g.graph().height * initialScale + 40);
+            } else {
+                svg.attr('height', 40);
+            }
+        }
+    }
+
+    render() {
+        return (
+            <Grid container spacing={3} >
+                <Grid item xs={12}>
+                    <Title>Task</Title>
+                </Grid>
+                <Grid item xs={12} md={6}>
+                    <Paper>
+                        <Grid item xs={12}>
+                            <svg
```

Comment:
Better to create a separate `TaskTileableDetail` class and make `TaskTileableChart` as a child element.